packages/medusa/src/api-v2/utils/middlewares/common/ensure-pub-key-sales-channel-match.ts

```diff
@@ -0,0 +1,52 @@
+import { NextFunction } from "express"
+import { MedusaRequest, MedusaResponse } from "../../../../types/routing"
+import { StoreCreateCartType } from "../../../store/carts/validators"
+
+/**
+ * If a publishable key (PK) is passed in the header of the request AND
+ * the request carries a sales channel id param in the url or body,
+ * we check if the sales channel is valid for the key.
+ *
+ * If the request does not carry a sales channel id, we attempt to assign
+ * a sales channel associated with the PK.
+ *
+ * @throws If sales channel id is passed as a url or body param
+ *         but that id is not in the scope defined by the PK from the header.
+ *         If the PK is associated with multiple sales channels but no
+ *         sales channel id is passed in the request.
+ */
+export async function ensurePublishableKeyAndSalesChannelMatch(
+  req: MedusaRequest<StoreCreateCartType> & { publishableApiKeyScopes },
+  res: MedusaResponse,
+  next: NextFunction
+) {
+  const pubKey = req.get("x-publishable-api-key")
+
+  if (pubKey) {
```

Comment:
`if (!pubKey)` so we can return early

Reply:
We never enter the `if` so we are, effectively, returning early if it is not there. Do you want this for readability?

Reply:
@olivermrbl yeah for readability, it reduces the nesting of the code